library/SimplePie/Parser.php

```diff
@@ -164,12 +164,30 @@ public function parse(&$data, $encoding, $url = '')
 			xml_set_element_handler($xml, 'tag_open', 'tag_close');
 
 			// Parse!
-			if (!xml_parse($xml, $data, true))
+			$wrapper = @is_writable(sys_get_temp_dir()) ? 'php://temp' : 'php://memory';
```

Comment:
Can be simplified by always using `php://memory` at the cost of not being able to parse just as large feeds